markets/perps-market/contracts/storage/PerpsMarket.sol

```diff
@@ -144,6 +153,16 @@ library PerpsMarket {
         self.size = (self.size + MathUtil.abs(newPosition.size)) - MathUtil.abs(oldPositionSize);
         self.skew += newPosition.size - oldPositionSize;
         oldPosition.updatePosition(newPosition);
+        int128 sizeDelta = newPosition.size - oldPositionSize;
+        // TODO add current market debt
+        emit MarketUpdated(
+            self.id,
+            self.skew,
+            self.size,
+            sizeDelta,
+            currentFundingRate(self),
+            currentFundingVelocity(self)
```

Comment:
randomly thought of this but `currentFundingRate` also calls `currentFundingVelocity` in its calculation so we're doing double the calcs here.

You can probably just use `self.lastFundingRate` for `currentFundingRate` since within the same transaction it was updated to the current block timestamp, the velocity won't have any impact

Reply:
Fixed